iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -535,19 +531,45 @@ public void testCTASFromHiveTable() {
         TableProperties.DEFAULT_FILE_FORMAT, fileFormat));
 
     List<Object[]> objects = shell.executeStatement("SELECT * FROM target ORDER BY id");
-    Assert.assertEquals(2, objects.size());
-    Assert.assertArrayEquals(new Object[]{1L, "Mike", "HR"}, objects.get(0));
-    Assert.assertArrayEquals(new Object[]{2L, "Linda", "Finance"}, objects.get(1));
+    HiveIcebergTestUtils.validateData(HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS,
+        HiveIcebergTestUtils.valueForRow(HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA, objects), 0);
+  }
+
+  @Test
+  public void testCTASPartitionedFromHiveTable() throws TException, InterruptedException {
+    Assume.assumeTrue(HiveIcebergSerDe.CTAS_EXCEPTION_MSG, testTableType == TestTables.TestTableType.HIVE_CATALOG);
+
+    shell.executeStatement("CREATE TABLE source (id bigint, name string) PARTITIONED BY (dept string) STORED AS ORC");
+    shell.executeStatement(testTables.getInsertQuery(
+        HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS, TableIdentifier.of("default", "source"), false));
+
+    shell.executeStatement(String.format(
+        "CREATE TABLE target PARTITIONED BY (dept, name) " +
+        "STORED BY '%s' TBLPROPERTIES ('%s'='%s') AS SELECT * FROM source",
```

Comment:
nit: after @lcspinter's patch we can use `STORED BY ICEBERG`

Reply:
Done